core/shared/src/main/scala/org/typelevel/log4cats/ErrorLogger.scala

```diff
@@ -28,6 +28,14 @@ trait ErrorLogger[F[_]] {
 }
 
 object ErrorLogger {
+  trait Fallback[F[_]] extends ErrorLogger[F] { _: MessageLogger[F] =>
```

Comment:
I'm having a hard time understanding what the purpose of `Fallback[F[_]]` is.

Is it a convenience mixin that ignores any `Throwable`s in the signatures of `ErrorLogger[F[_]]`?

Is it intended for end-users?

Seeing how it's used here I'd rather have it just be an implementation detail of the `SelfAwareStructuredLogger.liftF` because otherwise it increases the API surface area of the library quite a lot 😅

Reply:
Good question. I feel like it's a useful thing to have available but I don't have a very strong, specific answer to your question so yes, probably best to leave it as an implementation detail for now at least.

Reply:
I'd prefer making all the helper traits `private[log4cats]` (or whatever is best for bincompat, I am really bad at this part :grimacing:). We can always make them public later with little effort and in a minor version.

Reply:
`private[log4cats]` doesn't really protect bincompat, but it protects source compat, and we can grant a MiMa waiver if we're sure nothing else used it.

In multi-repo projects like http4s, it's unsafe to waive away package privates, because usages could be in another repo.  Here, it's probably safe: nothing other than this project should be in the `log4cats` namespace.

Either way, I'd agree locking it down is safer for libraries this low in the hierarchy.